6.0.0-alpha.13 (#7305)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.0.0-alpha.13

_Dec 23, 2022_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

- 🚀 New column menu design and API

> - 🌍 Update Korean and Russian locales

Reviewer: Be consistent with master release text may be
```suggestion
- 🌍 Improve Russian (ru-RU) and Korean (ko-KR) locales
```

- 📚 Documentation improvements
- 🐞 Bugfixes

### `@mui/x-data-grid@6.0.0-alpha.13` / `@mui/x-data-grid-pro@6.0.0-alpha.13` / `@mui/x-data-grid-premium@6.0.0-alpha.13`

#### Breaking changes

> - Rename column menu props and interfaces for consistency (#6619) MBilalShafi
> 
> - The `currentColumn` prop passed to `components.ColumnMenu` was renamed to `colDef`.
> - The `column` prop passed to the items of the column menu was renamed to `colDef`.
> - The `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` export has been removed.
> 
>     The following components and interfaces were been renamed for consistency:
> 

Reviewer: Organize them topic based may be?
```suggestion
New column menu design and API (#6619) @MBilalShafi

- The `currentColumn` prop passed to `components.ColumnMenu` was renamed to `colDef`.
- The `column` prop passed to the items of the column menu was renamed to `colDef`.
- The `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` export has been removed.
- The following components and interfaces were been renamed for consistency:

```

- 🐞 Bugfixes

### `@mui/x-data-grid@6.0.0-alpha.13` / `@mui/x-data-grid-pro@6.0.0-alpha.13` / `@mui/x-data-grid-premium@6.0.0-alpha.13`

#### Breaking changes

From https://github.com/mui/mui-x/pull/6619

- `column` and `currentColumn` recieved by `<GridColumnMenu />` and column menu items was renamed to `colDef`.
- `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` export has been removed.

- Some of the components have been renamed for consistency.

  **Community Package:**

  ```diff
  -<GridFilterMenuItem />
  +<GridColumnMenuFilterItem />
  ```

  ```diff
  -<HideGridColMenuItem />
  +<GridColumnMenuHideItem />
  ```

  ```diff
  -<GridColumnsMenuItem />
  +<GridColumnMenuColumnsItem />
  ```

  ```diff
  -<SortGridMenuItems />
  +<GridColumnMenuSortItem />
  ```

  ```diff
  -interface GridFilterItemProps
  +interface GridColumnMenuItemProps
  ```

  **Pro package:**
  ```diff
  -<GridColumnPinningMenuItems />
  +<GridColumnMenuPinningItem />
  ```

  **Premium package:**
  ```diff
  -<GridAggregationColumnMenuItem />
  +<GridColumnMenuAggregationItem />
  ```

  ```diff
  -<GridRowGroupingColumnMenuItems />
  -<GridRowGroupableColumnMenuItems />
  +<GridColumnMenuGroupingItem />
  ```

> From https://github.com/mui/mui-x/pull/7224

Reviewer: I usually used to remove these `From *` and just listed down all the breaking changes.

But in this specific use case, as we have breaking changes from multiple PRs, probably something like this may make more sense?

```
PR 1 Name and Details (#123) @Author

- Breaking change 1
- Breaking change 2
    - Sub item for breaking change 2
- Breaking change 3

PR 2 Name and Details (#456) @Author

- Breaking change 1
- Breaking change 2
- Breaking change 3
  - Sub item for breaking change 3
```

> From https://github.com/mui/mui-x/pull/7224

Author: Alright. thanks, @MBilalShafi, and @m4theushw for clearing it up. I've updated it so if you can take one last look that would be great.

- The `GridColumns` type was removed. Use `GridColDef[]` instead.
- The `GridActionsColDef` interface was removed. Use `GridColDef` instead.
- The `GridEnrichedColDef` type was removed. Use `GridColDef` instead.
- The `GridStateColDef` type was removed.
  If you use it to type `searchPredicate` - use `GridColumnsPanelProps['searchPredicate']` instead.
  If you use it to type `getApplyFilterFn` - use `GridFilterOperator['getApplyFilterFn']` instead.

From https://github.com/mui/mui-x/pull/7304

- The `GridDensityTypes` enum was removed. Use `GridDensity` type instead.

From https://github.com/mui/mui-x/pull/7067

- Remove column separator.
- Remove unused `disableHeaderSeparator`.

#### Changes

- [DataGrid] Allow disabling of buttons in column panel (#6947) @MBilalShafi
- [DataGrid] Improve column definition typing (#7224) @cherniavskii

> - [DataGrid] Improved column menu design and api (#6619) @MBilalShafi

Reviewer 2: ```suggestion
- [DataGrid] Improve column menu design and API (#6619) @MBilalShafi
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.0.0-alpha.13

_Dec 23, 2022_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

- 🚀 Improved column menu design and API

> 

Reviewer: ```suggestion
- 🌍 Improve Korean and Russian locales
- 📚 Documentation improvements
- 🐞 Bugfixes

```

From https://github.com/mui/mui-x/pull/7304

> #### Breaking changes

Author: Should be ok now.

- [x] Remove unused `disableHeaderSeparator`

## Design
- [Figma Process](https://www.figma.com/file/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=1227%3A62449&t=MAO4kO2qvUssZFTG-0)
- Figma Final
From https://github.com/mui/mui-x/pull/7249

#### Breaking changes

- `AdapterLuxon` now required `luxon` in version `3.0.2` or higher in order to work.

  Take a look at the [Upgrading Luxon](https://moment.github.io/luxon/#/upgrading) guide if you are using an older version.

From https://github.com/mui/mui-x/pull/6619

- New column menu design with improved look and feel
- Ability to add new menu items and place them on any order
- Replaced all the items to be injected via pipe processor for better control through respective hooks
- Added ability to override/reorder column menu items which may help add more items in any order e.g. for https://github.com/mui/mui-x/issues/5307
- Ability to customize the menu by passing a component in column menu slot
- New l10n labels added
- Added dedicated component and icon slots for Pro and Premium packages for shipping only required components per package
- Fix accessibility bug of focus shifting to body when an item gets hidden

### Breaking changes

> - `column` and `currentColumn` is now renamed to `colDef`

Reviewer: ```suggestion
- `column` and `currentColumn` recieved by `<GridColumnMenu />` and column menu items was renamed to `colDef`
```

- `DATA_GRID_DEFAULT_SLOTS_COMPONENTS` export has been removed

- Some of the components have been renamed for consistency.

  **Community Package:**
  ```diff
  - <GridFilterMenuItem />
  + <GridColumnMenuFilterItem />
  ```

  ```diff
  - <HideGridColMenuItem />
  + <GridColumnMenuHideItem />
  ```

  ```diff
  - <GridColumnsMenuItem />
  + <GridColumnMenuColumnsItem />
  ```

  ```diff
  - <SortGridMenuItems />
  + <GridColumnMenuSortItem />
  ```

  ```diff
  - interface GridFilterItemProps
  + interface GridColumnMenuItemProps
  ```

  **Pro package:**
  ```diff
  - <GridColumnPinningMenuItems />
  + <GridColumnMenuPinningItem />
  ```

  **Premium package:**
  ```diff
  - <GridAggregationColumnMenuItem />
  + <GridColumnMenuAggregationItem />
  ```

  ```diff
  - <GridRowGroupingColumnMenuItems />
  - <GridRowGroupableColumnMenuItems />
  + <GridColumnMenuGroupingItem />
  ```

> 
> ## Design
> - [Figma Process](https://www.figma.com/file/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=679%3A38262&t=aFUzX2phxDdIDwFL-0)
> - [Final](https://www.figma.com/proto/W6TItYkiGJZjL5zLnfeSjy/Data-grid?node-id=1239%3A64017&scaling=min-zoom&page-id=679%3A38262)

Reviewer: ```suggestion
```
